keel-core/src/main/kotlin/com/netflix/spinnaker/keel/resources/ResourceSpecIdentifier.kt

```diff
@@ -16,22 +15,19 @@ import org.springframework.stereotype.Component
  */
 @Component
 class ResourceSpecIdentifier(
-  private val kinds: List<SupportedKind<*>>
+  private val kinds: List<SupportedKind<*>>,
+  private val extensionRegistry: ExtensionRegistry? = null
 ) {
-  private val log by lazy { LoggerFactory.getLogger(javaClass) }
-
   @Autowired
-  constructor(extensionRegistry: ExtensionRegistry) :
-    this(extensionRegistry
-      .extensionsOf<ResourceSpec>()
+  constructor(extensionRegistry: ExtensionRegistry) : this(emptyList(), extensionRegistry)
+
+  private val ExtensionRegistry.supportedKinds: List<SupportedKind<*>>
+    get() = extensionsOf<ResourceSpec>()
       .entries
       .map { SupportedKind(ResourceKind.parseKind(it.key), it.value) }
-    ) {
-    log.debug("Registered the following kinds from extension registry: ${kinds.map { it.kind }.joinToString()}")
-  }
 
   fun identify(kind: ResourceKind): Class<out ResourceSpec> =
-    kinds.find { it.kind == kind }?.specClass ?: throw UnsupportedKind(kind)
+    (extensionRegistry?.supportedKinds ?: kinds).find { it.kind == kind }?.specClass ?: throw UnsupportedKind(kind)
```

Comment:
This gives priority to the extension registry which is auto-wired by default, and falls back on the list of explicit supported kinds which is only used in tests.

Reply:
I recommend adding this information as a comment

Reply:
Done!